sprint-2/step-1

Ruslick opened the pull request:
refactor fetch query, refactor data managment to context, update reducerBC, rename component ingList to ingCategory

Ruslick committed:
fetch, data managment, reducerBC, ingridientCategory

refactor fetch query, refactor data managment to context, update reducerBC, rename component ingList to ingCategory

gennady-bars commented on src/utils/constants.js:
```diff
@@ -1,3 +1,9 @@
-const URL = "https://norma.nomoreparties.space/api/ingredients";
+const URL = "https://norma.nomoreparties.space";
```
Отлично, что базовый урл вынесен в константу

gennady-bars commented on src/utils/ingridientsContextApi.js:
```diff
@@ -0,0 +1,5 @@
+import { createContext } from "react";
+
+const ingridientsContextApi = createContext(null);
```
`ingridientsContextApi` переменная для контекста должна быть с большой буквы `IngridientsContextApi`, так как она выступает в роли компонента реакта в верстке

gennady-bars commented on src/utils/orderContext.js:
```diff
@@ -0,0 +1,5 @@
+import { createContext } from "react";
+
+const orderContext = createContext(null);
```
`orderContext` переменная для контекста должна быть с большой буквы `ОrderContext`, так как она выступает в роли компонента реакта в верстке

gennady-bars commented on src/utils/postOrder.js:
```diff
@@ -0,0 +1,38 @@
+	};
+
+	return fetch(ORDERS_URL, getOptions())
+		.then((res) => {
```
```js
.then(res => {
                if (res.ok) {
                    return res.json();
                }
                return Promise.reject(`Ошибка ${res.status}`);
            })

```

У Вас повторяется код **проверки ответа** во всех запросах. Чтобы этого избежать, нужно сделать отдельный метод `checkResponse`, который будет проверять ответ от сервера:

```js
function checkResponse(res) {
        // тут проверка ответа
    }

```

и теперь можно удалить кучу ненужно повторного кода и заменить его этим:

```js
 .then(checkResponse)

```

Обратите внимание, что передается только **ссылка** на метод. Не нужно его вызывать. Он сам вызовется, так как в `then` нужно передавать именно **функцию**, а не **вызов** функции.

gennady-bars commented on src/utils/getIngridients.js:
```diff
@@ -0,0 +1,19 @@
+export const getIngridients = () => {
+	return fetch(INGRIDIENTS_URL)
+		.then((res) => {
+			if (res.ok) {
```
У Вас повторяется код **проверки ответа** во всех запросах. Чтобы этого избежать, нужно сделать отдельный метод `checkResponse`, который будет проверять ответ от сервера:

gennady-bars commented on src/utils/getIngridients.js:
```diff
@@ -0,0 +1,19 @@
+			}
+			throw error;
+		})
+		.then((responseData) => {
```
и проверка на `success` тоже повторяется во всех запросах. Лучше сделать `checkSuccess`

gennady-bars commented on src/components/App/App.jsx:
```diff
@@ -2,31 +2,42 @@ import styles from "./App.module.css";
-			<div className="container">
-				<section className={styles.wrapper}>
-					{data ? (
+			<ingridientsContextApi.Provider value={{ data }}>
```
Отличный контекст на все приложение

gennady-bars commented on src/components/App/App.jsx:
```diff
@@ -2,31 +2,42 @@ import styles from "./App.module.css";
+			<ingridientsContextApi.Provider value={{ data }}>
+				<div className="container">
+					<section className={styles.wrapper}>
 						<>
```
`<>` пустые теги нужны только тогда, когда нет **общего** тега (компонента), который оборачивает всю верстку

gennady-bars reviewed (commented):
Здравствуйте. (**Нужно развернуть общий комментарий ↓**)

Работа проделана огромная


* Отлично, что базовый урл вынесен в константу
* Отличный контекст на все приложение

но есть некоторые недочеты: 

* `ingridientsContextApi` переменная для контекста должна быть с большой буквы `IngridientsContextApi`, так как она выступает в роли компонента реакта в верстке
* У Вас повторяется код **проверки ответа** во всех запросах. Чтобы этого избежать, нужно сделать отдельный метод `checkResponse`, который будет проверять ответ от сервера:
* `<>` пустые теги нужны только тогда, когда нет **общего** тега (компонента), который оборачивает всю верстку



**Можно лучше**

* Лучше ограничивать кол-во срабатываний фильтраций массивов при помощи хука `useMemo`.  Это касается таких операций, как `filter`, `map`, `forEach`, `reduce` и тд . Подробнее https://ru.reactjs.org/docs/hooks-reference.html#usememo 
* При кликах на табы `Булки, Соусы, Начинки` можно сделать логику **`плавного`** скролла до секции. Это можно сделать с помощью хука `useRef` и методом ` scrollIntoView`. Нужно установить для 3х секций 3 `useRef` и следить за изменением табов при кликах на них через функцию клика. Как меняется таб, то запускается ` scrollIntoView ` со значением ` behavior: "smooth"`
Подробнее про `useRef` тут https://ru.reactjs.org/docs/hooks-reference.html#useref
Подробнее про ` scrollIntoView` тут https://developer.mozilla.org/ru/docs/Web/API/Element/scrollIntoView

Исправьте, пожалуйста, недочеты и работа будет принята. Пожалуйста, проверьте **работоспособность проекта** и наличие возможных ошибок в консоли браузера (кнопка `F12`) перед отправкой на ревью. 

Напоминаю, что работа может быть принята только после исправления всех критических замечаний `Нужно исправить`.

Удачного рефакторинга кода.

Ruslick committed:
context, request, useMemo, scroll

rename context files, merge request hanlers, add useMemo to arrays methods, add tabs scroll to burgerIngridients

gennady-bars reviewed (commented):
К сожалению, работу придется отклонить без потери итерации, так как проект не запускается. Скрин ошибки http://joxi.ru/BA0YjEvS1RLaXr


Видимо, **кэш гита** не отслеживает регистр файлов. Для гита без разницы, с маленькой буквы или с большой сделан файл. Он считает их одинаковыми. Попробуйте очистить кэш гита. Посоветуйтесь с наставником или посмотрите по этой ссылке, как это сделать https://coderoad.ru/41863484/%D0%9E%D1%87%D0%B8%D1%81%D1%82%D0%B8%D1%82%D1%8C-%D0%BB%D0%BE%D0%BA%D0%B0%D0%BB%D1%8C%D0%BD%D1%8B%D0%B9-%D0%BA%D1%8D%D1%88-git

Ruslick committed:
clear ceche

gennady-bars reviewed (approved):
Поздравляю! Ваша работа принята.

Вы отлично потрудились.

Удачного дальнейшего обучения.
